docs/rst/tutorials/micro/custom_types/custom_types.rst

```diff
@@ -0,0 +1,424 @@
+.. _tutorials_micro_custom_types:
+
+Creating custom types
+=====================
+
+.. contents:: Table of Contents
+    :depth: 1
+    :local:
+    :backlinks: none
+
+
+This tutorial aims at providing step-by-step guidance for those users interested in adding custom message definition to the micro-ROS build. Those instructions can be used to extend the type definition for topics, services and actions provided by ROS 2 and micro-ROS.
+
+.. TODO(pgarrido): explain how to integrate in build systems
```

Comment:
can be done on this PR, or is it going to be in a new tutorial? is it the line 21 link?

Reply:
aa6c6df32fcb0e537b016fdfa4e81f437e518598

Reply:
aa6c6df32fcb0e537b016fdfa4e81f437e518598